layerfetcher/layers.go

```diff
@@ -337,11 +337,16 @@ func (l *Logic) fetchLayerData(ctx context.Context, logger log.Log, layerID type
 	}
 	l.mutex.Unlock()
 
+	l.fetcher.TrackBallotsPeer(ctx, peer, ballotsToFetch)
```

Comment:
do this in `receiveLayerContent` after `extract extractPeerResult()`
there you should just gather a big list of hashes (ballots/blocks) and register that peer with those hashes.
the API in fetch for registering hashes does not need to know whether what data type this hash is.

Reply:
done, thanks